src/Features/LanguageServer/Protocol/Handler/Completion/CompletionResolveData.cs

```diff
@@ -13,5 +14,7 @@ internal class CompletionResolveData
         public Position Position { get; set; }
 
         public string DisplayText { get; set; }
+
+        public CompletionTrigger CompletionTrigger { get; set; }
```

Comment:
Anyone know whether it's OK to pass Roslyn types to LSP? CompletionTrigger is a struct defined in Roslyn.
Alternatively, instead of using CompletionTrigger here, we could instead pass in the trigger character and LSP trigger kind, but this would later require a conversion and the allocation of a CompletionTrigger object in CompletionResolveHandler.

Reply:
Anything you pass just gets serialized, and would need to be deserialized on the other end, so if the structs are the same shape and the same DataContract attributes it would work... but I'm guessing this doesn't have that :)

Reply:
Yeah it needs to be serializable - it looks like it's just an enum and a char and streamjsonrpc might just serialize it.

Note that this does not need to be de-serialized on the client, it's pass through back to the server to de-serialize on resolve.

Reply:
Hmm I debugged through and it does look like the resolve handler is being deserializing the enum/char fine... so hopefully this approach is OK?

Reply:
The only risk I see is if someone adds a new property that is not easily serializable or de serializable.  Because of that I usually define my own types to avoid that risk, but I'm open to other opinions.

Reply:
Can we possibly make a note in CompletionTrigger's summary saying that if a new property is added, it should be serializable or LSP needs to be updated?
I think we have something similar for semantic tokens where we include a note saying that if the type/class is updated, LSP should also be updated.

Reply:
that is OK with me